validator/validators/feature/feature_validator.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package feature
+
+import (
+	"time"
+
+	"go.uber.org/multierr"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/awsservice"
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/validator/models"
+)
+
+type FeatureValidator struct {
+	vConfig models.ValidateConfig
+}
+
+var _ models.ValidatorFactory = (*FeatureValidator)(nil)
+
+func NewFeatureValidator(vConfig models.ValidateConfig) models.ValidatorFactory {
+	return &FeatureValidator{
+		vConfig: vConfig,
+	}
+}
+
+func (s *FeatureValidator) GenerateLoad() error {
+	var (
+		multiErr              error
+		dataRate              = s.vConfig.GetDataRate()
+		agentCollectionPeriod = s.vConfig.GetAgentCollectionPeriod()
+		agentConfigFilePath   = s.vConfig.GetCloudWatchAgentConfigPath()
+		receiver              = s.vConfig.GetPluginsConfig()
+	)
+
+	err := common.StartLogWrite(agentConfigFilePath, agentCollectionPeriod, dataRate)
+	if err != nil {
+		multiErr = multierr.Append(multiErr, err)
+	}
+
+	// Sending metrics based on the receivers; however, for scraping plugin  (e.g prometheus), we would need to scrape it instead of sending
+	err = common.StartSendingMetrics(receiver, agentCollectionPeriod, dataRate)
+	if err != nil {
+		multiErr = multierr.Append(multiErr, err)
+	}
+
+	return multiErr
+}
+
+func (s *FeatureValidator) CheckData(startTime, endTime time.Time) error {
+	var (
+		multiErr error
+	)
+
+	return multiErr
+}
+
+func (s *FeatureValidator) Cleanup() error {
+	var (
+		dataType      = s.vConfig.GetDataType()
+		ec2InstanceId = awsservice.GetInstanceId()
+	)
+	switch dataType {
+	case "logs":
+		awsservice.DeleteLogGroup(ec2InstanceId)
+	}
```

Comment:
I saw you mentioned 
> generate load in this PR with metrics and logs only

in the previous [comment](https://github.com/aws/amazon-cloudwatch-agent-test/pull/162/files#r1132928801). Is that because we cannot really cleanup metrics so that we don't have anything here for metrics?
Also, since this is in a switch-case block, are we expecting to see other dataType that need to be cleaned up here?

Reply:
```
 Is that because we cannot really cleanup metrics so that we don't have anything here for metrics?
```
Yes. If there is one for cleaning metrics/namespace, then I would love to add that. Unfortunately, we don't have that.


```
Also, since this is in a switch-case block, are we expecting to see other dataType that need to be cleaned up here?
```
I don't think there is another use case for this since trace does not [support deleting ](https://repost.aws/questions/QUevPeGFTNQoaZHFQufZTBrg/is-there-any-way-to-delete-traces-in-x-ray-to-save-money) but only filtering. I'm fine with add a CleanUp function for only `DeleteLogGroup`  here but if anyone has a urge against it, then I'm good with delting it too.